WeatherApp/WeatherApp/HomeScreen/HomeScreenViewModel.swift

```diff
@@ -0,0 +1,30 @@
+import SwiftUI
+
+class HomeScreenViewModel: ObservableObject {
+
+    private let apiKey = "ff4cd4d2c654b4100a2712f4cbaeb732"
+    @Published var weather: WeatherModel?
+
+    func fetchWeatherZagreb() async {
+        var urlComponents = URLComponents(string: "https://api.openweathermap.org/data/2.5/weather")!
+        urlComponents.queryItems = [
+            URLQueryItem(name: "q", value: "Zagreb"),
+            URLQueryItem(name: "appid", value: apiKey),
+            URLQueryItem(name: "units", value: "metric")
+        ]
+
+        guard let url = urlComponents.url else { return }
+
+        do {
+            let (data, _) = try await URLSession.shared.data(from: url)
+            let decodedData = try JSONDecoder().decode(WeatherModel.self, from: data)
+            DispatchQueue.main.async {
+                self.weather = decodedData
```

Comment:
You have created a memory leak here. You are strongly capturing self within a closure which would result in ARC not releasing this object from memory.

```swift
            DispatchQueue.main.async { [weak self] in
                 self?.weather = decodedData
             }
```

Read a bit about ARC and swift memory management. About memory leaks and how to spot them.
Rule of thumb is -> whenever you are referencing self from within a closure (usually compiler is the one that will throw you a warning regarding "you have to wirte self here...") you are in danger of creating a memory leak.